exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -684,60 +669,214 @@ func (ku *Kucoin) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Subm
 	if err != nil {
 		return nil, err
 	}
+	var o string
 	switch s.AssetType {
 	case asset.Futures:
-		o, err := ku.PostFuturesOrder(ctx, &FuturesOrderParam{
+		if s.Leverage == 0 {
+			s.Leverage = 1
+		}
+		var stopOrderType, stopOrderBoundary string
+		switch s.Type {
+		case order.Stop, order.StopLimit, order.TrailingStop:
+			if s.TriggerPrice == 0 {
+				break
+			}
+			switch s.TriggerPriceType {
+			case order.IndexPrice:
+				stopOrderType = "IP"
+			case order.MarkPrice:
+				stopOrderType = "MP"
+			case order.LastPrice:
+				stopOrderType = "TP"
+			}
+			switch s.Type {
+			case order.StopLimit:
+				switch s.Side {
+				case order.Sell:
+					stopOrderBoundary = "up"
+				case order.Buy:
+					stopOrderBoundary = "down"
+				}
+			case order.TrailingStop, order.Stop:
+				switch s.Side {
+				case order.Sell:
+					// Stop-loss when order type is order.Stop
+					stopOrderBoundary = "down"
+				case order.Buy:
+					// Take Profit when order type is order.Stop
+					stopOrderBoundary = "up"
+				}
+			}
+		}
+		o, err = ku.PostFuturesOrder(ctx, &FuturesOrderParam{
```

Comment:
I would like to see this part hit in tests please. Dont need coverage of each stopOrderType or stopOrderBoundary, just getting through asset.Futures to here